miniscope_io/models/devupdate.py

```diff
@@ -0,0 +1,123 @@
+"""
+Models for device update commands.
+"""
+
+from enum import Enum
+
+import serial.tools.list_ports
+from pydantic import BaseModel, ConfigDict, field_validator, model_validator
+
+
+class DeviceCommand(Enum):
+    """Commands for device."""
+
+    REBOOT = 200
+
+
+class UpdateCommandDefinitions:
+    """
+    Definitions of Bit masks and headers for remote update commands.
+    """
+
+    # Header to indicate target/value.
+    # It probably won't be used in other places so defined here.
+    id_header = 0b00000000
+    target_header = 0b11000000
+    LSB_header = 0b01000000
+    MSB_header = 0b10000000
+    LSB_value_mask = 0b000000111111  # value below 12-bit
+    MSB_value_mask = 0b111111000000  # value below 12-bit
+    reset_byte = 0b11111111
+
+
+class UpdateTarget(Enum):
+    """
+    Targets to update. Needs to be under 6-bit.
+    """
```

Comment:
we will eventually want to unify this with a device configuration, since I assume these values won't be static for all miniscopes for all time - so e.g. we have a common set of targetable values and then each device provides a mapping to the int code/a method for setting each of them. 

fine for now tho, since the ABC refactor should actually give us a reasonable place to put these things

Reply:
Kind of depends on how we want the interface to look because these have quite different staticness. So, suppose there is a GUI; the current device config (and maybe initial values for this update) should be used for initiating, and the stuff in this update should be handled in the interface.

Reply:
But anyways let's leave it for later